warehouse/query-core/src/main/java/datawave/query/iterator/QueryIterator.java

```diff
@@ -1293,9 +1297,17 @@ protected DocumentProjection getProjection() {
         DocumentProjection projection = new DocumentProjection(this.isIncludeGroupingContext(), this.isReducedResponse(), isTrackSizes());
         
         if (this.useWhiteListedFields) {
+            // make sure we including any fields being matched in the limit fields mechanism
+            if (!this.matchingFieldSets.isEmpty()) {
+                this.whiteListedFields.addAll(this.matchingFieldSets.stream().flatMap(s -> s.stream()).collect(Collectors.toList()));
+            }
             projection.setIncludes(this.whiteListedFields);
             return projection;
         } else if (this.useBlackListedFields) {
+            // make sure we including any fields being matched in the limit fields mechanism
```

Comment:
```suggestion
            // make sure we exclude any fields being matched in the limit fields mechanism
```

Reply:
Actuall the comment is correct, the code is incorrect here.

Reply:
So you want to add the fields being matched to the blacklisted fields?

Reply:
What change needs to be made?  I was just commenting on the grammar.

Reply:
You may have already updated the code, but I can't tell since you force pushed your changes.